sdk/src/utils/test.rs

```diff
@@ -371,6 +409,102 @@ impl crate::signer::RemoteSigner for TempRemoteSigner {
+        static START_CERTIFICATE: &str = "-----BEGIN CERTIFICATE-----";
+        static END_CERTIFICATE: &str = "-----END CERTIFICATE-----";
+        static START_KEY: &str = "-----BEGIN PRIVATE KEY-----";
+        static END_KEY: &str = "-----END PRIVATE KEY-----";
+
+        let mut name = alg.to_owned().to_uppercase();
+        name.insert(2, '-');
+
+        let key = key
+            .replace("\n", "")
+            .replace(START_KEY, "")
+            .replace(END_KEY, "");
+        let key = crate::utils::base64::decode(&key).unwrap();
+
+        let certs = cert
+            .replace("\n", "")
+            .replace(START_CERTIFICATE, "")
+            .split(END_CERTIFICATE)
+            .map(|x| crate::utils::base64::decode(x).unwrap())
+            .collect();
+
+        Self {
+            signing_alg: alg.parse().unwrap(),
+            signing_alg_name: name,
+            certs,
+            key,
+        }
+    }
+}
+
+#[cfg(target_arch = "wasm32")]
+#[async_trait::async_trait(?Send)]
+impl crate::signer::AsyncSigner for WebCryptoSigner {
+    fn alg(&self) -> SigningAlg {
+        self.signing_alg
+    }
+
+    fn certs(&self) -> Result<Vec<Vec<u8>>> {
+        Ok(self.certs.clone())
+    }
+
+    async fn sign(&self, claim_bytes: Vec<u8>) -> crate::error::Result<Vec<u8>> {
+        use js_sys::{Array, Object, Reflect, Uint8Array};
+        use wasm_bindgen_futures::JsFuture;
+        use web_sys::CryptoKey;
+
+        use crate::wasm::context::WindowOrWorker;
+        let context = WindowOrWorker::new().unwrap();
+        let crypto = context.subtle_crypto().unwrap();
+
+        let mut data = claim_bytes.clone();
+        let promise = crypto
+            .digest_with_str_and_u8_array("SHA-256", &mut data)
+            .unwrap();
+        let result = JsFuture::from(promise).await.unwrap();
+        let mut digest = Uint8Array::new(&result).to_vec();
+
+        let key = Uint8Array::new_with_length(self.key.len() as u32);
+        key.copy_from(&self.key);
+        let usages = Array::new();
+        usages.push(&"sign".into());
+        let alg = Object::new();
+        Reflect::set(&alg, &"name".into(), &"ECDSA".into()).unwrap();
+        Reflect::set(&alg, &"namedCurve".into(), &"P-256".into()).unwrap();
+
+        let promise = crypto
+            .import_key_with_object("pkcs8", &key, &alg, true, &usages)
+            .unwrap();
+        let key: CryptoKey = JsFuture::from(promise).await.unwrap().into();
+
+        let alg = Object::new();
+        Reflect::set(&alg, &"name".into(), &"ECDSA".into()).unwrap();
+        Reflect::set(&alg, &"hash".into(), &"SHA-256".into()).unwrap();
+        let promise = crypto
+            .sign_with_object_and_u8_array(&alg, &key, &mut digest)
+            .unwrap();
+        let result = JsFuture::from(promise).await.unwrap();
+        Ok(Uint8Array::new(&result).to_vec())
+    }
+
+    fn reserve_size(&self) -> usize {
+        10000
```

Comment:
How do you know this size is sufficient.  The size need to cover the size of certificate chain + size of timestamp response + size of OCSP response + size of structure.  If the size if not sufficient the code will throw an error.

Reply:
If this is just test code then the fixed size if fine.

Reply:
This is just test code.